python_modules/dagster-graphql/dagster_graphql/schema/inputs.py

```diff
@@ -137,9 +137,36 @@ class Meta:
         name = "PartitionSetSelector"
 
 
+class GraphenePartitionRangeSelector(graphene.InputObjectType):
+    start = graphene.NonNull(graphene.String)
+    end = graphene.NonNull(graphene.String)
+
+    class Meta:
+        description = """This type represents a partition range selection with start and end."""
+        name = "PartitionRangeSelector"
+
+
+class GraphenePartitionsSelector(graphene.InputObjectType):
+    range = graphene.NonNull(GraphenePartitionRangeSelector)
```

Comment:
Looks like we're allowing a maximum of 1 range per asset.

In this case, I think we can remove the `GraphenePartitionsSelector` object and just pass a `GraphenePartitionRangeSelector` into `GraphenePartitionsByAssetSelector`.

Reply:
Thoughts on renaming `GraphenePartitionRangeSelector` to `GraphenePartitionRangeSelection`?

Reply:
Hi @clairelin135, yes, ideally we would not allow more than 1 range per asset for backfill. 

However, the implementation here is based on an earlier design proposal from @sryza , unless there is a specific reason for it, I am happy to flatten it out :)

Reply:
@clairelin135, is my understanding correct that this would make it look like 
```
      "partitions": {
        "start": "2022-02-01",
        "end": "2022-02-04",
      }
```

instead of 

```
      "partitions": {
        "range": {
          "start": "2022-02-01",
          "end": "2022-02-04",
        }
      }
```

My thinking here was that the latter gives us the freedom to allow other ways of specifying sets of partitions in the future.  For example:

```
      "partitions": {
        "keys": ["north_america", "asia"]
      }
```

Reply:
ok, that sounds reasonable to me. Wasn't sure initially why we wanted two layers of objects